addons/stock/models/stock_warehouse.py

```diff
@@ -148,6 +148,12 @@ def write(self, vals):
                 for inter_wh_route in to_remove_routes:
                     warehouse.write({'route_ids': [(3, inter_wh_route.id)]})
 
+        # check warehouse is archived related operation types and location should archived
+        if 'active' in vals:
+            operation_types = self.env['stock.picking.type'].search([('warehouse_id', 'in', self.ids), ('active', '!=', vals.get('active'))])
+            operation_types.write({'active': vals.get('active')})
+            (self.lot_stock_id + self.wh_input_stock_loc_id + self.wh_qc_stock_loc_id + self.wh_output_stock_loc_id + self.wh_pack_stock_loc_id).write({'active': vals.get('active')})
```

Comment:
Trying to read or write a field on multiple records will raise an error. Expected singleton .